### PR TITLE
Fix for missing icon, due to file system permissions

### PR DESCRIPTION
**Pull request by GurliGebis**
This changes the implementation to have NppShell.dll and NppShell.msix inside a subfolder, since having them in the same folder as notepad++.exe causes the filesystem permissions to get messed up, causing the icon to be missing when launched from the right click context menu.

This, along with changes to normal NPP repo fixes the https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13410 issue.

### CODE REVIEW

**Comment by GurliGebis** [comment]
@donho please don't merge until further notice.
I need to make a small fix to ensure upgrading from a version with the msix in the main folder works.
Unregistering does not remove the ACL, so I'll fix that, update this PR and let you know

**Comment by GurliGebis** [comment]
@donho this one is ready to merge when you get home - it now handles resetting the ACL on the Notepad++ folder.

File: AclHelper.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        free(emptyAcl);
+    }
+}
+
+DWORD AclHelper::ResetAcl(const wstring& path)
+{
+    if (emptyAcl)
+    {
+        return SetNamedSecurityInfoW(const_cast<LPWSTR>(path.c_str()), SE_FILE_OBJECT, DACL_SECURITY_INFORMATION | UNPROTECTED_DACL_SECURITY_INFORMATION, NULL, NULL, emptyAcl, NULL);
```
Comment (donho) [inline AclHelper.cpp:26]: Once we changed the security information to what we need, shouldn't we restore the default security information ?

Comment (GurliGebis) [reply AclHelper.cpp:26]: We reset it, so it inherits what is on the Program Files folder, which is the default behavior.

**Pull request closed without merging**